From a0664b15d42970ea72de67bdaf57671b7a95b11e Mon Sep 17 00:00:00 2001
From: Ben Dooks <ben.dooks@codethink.co.uk>
Date: Fri, 16 Jun 2023 15:44:00 +0100
Subject: [PATCH 2/3] zstd: fix g_debuglevel export warning

The g_debuglevel variable in debug.c is only used when DEBUGLEVEL is
defined to be above 2. This means by default there's no actual definition
of this in the headers, so sparse is giving the following warning:

lib/zstd/common/debug.c:24:5: warning: symbol 'g_debuglevel' was not declared. Should it be static?

We can use the same check as in the header to remove this if it isn't
going to be used, silencing the warning and removing a small bit of unused
data.

Signed-off-by: Ben Dooks <ben.dooks@codethink.co.uk>
Signed-off-by: Nick Terrell <terrelln@fb.com>
---
 lib/zstd/common/debug.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/zstd/common/debug.c b/lib/zstd/common/debug.c
index e56ff6464..d77926cba 100644
--- a/lib/zstd/common/debug.c
+++ b/lib/zstd/common/debug.c
@@ -22,4 +22,6 @@
 
 #include "debug.h"
 
+#if (DEBUGLEVEL>=2)
 int g_debuglevel = DEBUGLEVEL;
+#endif
-- 
2.43.0.232.ge79552d197

